File: io/io/src/test/resources/compat/issue-754/src/java/LegacyBinaryImplementation.java
package io.bluetape4k.io.serializer.compat.issue754.java;

import io.bluetape4k.io.serializer.BinarySerializer;
import java.nio.charset.StandardCharsets;

public final class LegacyBinaryImplementation implements BinarySerializer {
    public int serializeCalls;
    public int deserializeCalls;

    @Override
    public byte[] serialize(Object graph) {
        serializeCalls++;
        return graph == null ? new byte[0] : graph.toString().getBytes(StandardCharsets.UTF_8);
    }

    @Override
    @SuppressWarnings("unchecked")
    public <T> T deserialize(byte[] bytes) {
        deserializeCalls++;
        return bytes == null || bytes.length == 0
            ? null
            : (T) new String(bytes, StandardCharsets.UTF_8);
    }
}

File: io/io/src/test/resources/compat/issue-754/src/kotlin/LegacyBinaryImplementation.kt
package io.bluetape4k.io.serializer.compat.issue754.kotlin

import io.bluetape4k.io.serializer.BinarySerializer

class LegacyBinaryImplementation: BinarySerializer {
    override fun serialize(graph: Any?): ByteArray = graph?.toString()?.encodeToByteArray() ?: byteArrayOf()

    @Suppress("UNCHECKED_CAST")
    override fun <T: Any> deserialize(bytes: ByteArray?): T? =
        bytes?.takeIf { it.isNotEmpty() }?.decodeToString() as T?
}
